### PR TITLE
Reader P2: Follow/Unfollow comments capability 

### LINKED ISSUE #14574: Adding to the Reader P2 Follow/Unfollow comments capability
### Background

We already massively use P2 blogs internally in a8c to enforce teams project collaboration and we are going to launch the P2 Product [p7DVsv-8N3-p2] pretty soon. 

In a P2 web page we allow a user to Follow/Unfollow a stream of comments to a post by the `Follow` button available at the bottom of a post itself.

<p align="center">
  
</p>

The effect of Following a post comments stream is that the user gets notified by emails of new comments to that post.

### Task

With the above in mind we would like to add the capability in the app to Follow/Unfollow a P2 post comments stream from the Reader (will add some mocks related to this soon)

### Relevant endpoints

- [Subscribe to a post comments stream](https://developer.wordpress.com/docs/api/1.1/post/sites/%24site/posts/%24post/subscribers/new/)  
- [Unsubscribe from a post comments stream](https://developer.wordpress.com/docs/api/1.1/post/sites/%24site/posts/%24post/subscribers/mine/delete/)
- [Get subscription state](https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/posts/%24post/subscribers/mine/)

NOTE: the new P2 Product blogs are marked as `"is_wpforteams_site": true,` in the site options response

<p align="center">
  
</p>

as of current understanding this does not apply to older a8c internal P2s. In case we cannot find a flag that marks an old P2 site as being a P2 we could eventually evaluate to assume that the blogs that are in the reader Automattic Tab are P2s (note those P2s are only available to Automatticians).

<p align="center">
  
</p>

cc @leandroalonso

### PR DESCRIPTION
**Pull request by leandroalonso**
Fixes #14574

Adds follow/unfollow comments capability in Reader.

## To test

- [ ] #14655
- [ ] #14656
- [ ] #14703

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m [hunk 1]
```diff
@@ -287,6 +288,9 @@ - (void)configurePostHeader
     headerView.onClick = ^{
         [weakSelf handleHeaderTapped];
     };
+    headerView.onFollowConverationClick = ^{
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m]: Small typo fix. 

```suggestion
    headerView.onFollowConversationClick = ^{
```

File: WordPress/Classes/ViewRelated/Reader/ReaderPostHeaderView.h [hunk 2]
```diff
@@ -10,11 +10,22 @@ extern const CGFloat PostHeaderViewAvatarSize;
 typedef void (^ReaderPostHeaderCallback)(void);
 @property (nonatomic, copy) ReaderPostHeaderCallback onClick;
 
+/**
+ A ReaderPostHeaderFollowingCallback block to be executed whenever the user pressed this view.
+ */
+typedef void (^ReaderPostHeaderFollowingCallback)(void);
+@property (nonatomic, copy) ReaderPostHeaderFollowingCallback onFollowConverationClick;
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Reader/ReaderPostHeaderView.h]: Small typo fix: 

```suggestion
@property (nonatomic, copy) ReaderPostHeaderFollowingCallback onFollowConversationClick;
```

File: WordPress/Classes/ViewRelated/Reader/ReaderPostHeaderView.m [hunk 3]
```diff
@@ -195,6 +224,21 @@ - (void)setSubtitle:(NSString *)title
...
+    _isSubscribedToPost = isSubscribedToPost;
+    [self.followConversationButton setSelected:isSubscribedToPost];
+}
+
+#pragma mark - Button Action Helpers
+
+- (void)followConversationButtonTapped
+{
+    if (self.onFollowConverationClick) {
+        self.onFollowConverationClick();
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Reader/ReaderPostHeaderView.m]: Same typo as above
```suggestion
    if (self.onFollowConversationClick) {
        self.onFollowConversationClick();
```

File: WordPress/WordPressTest/FollowCommentsServiceTests.swift [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    override func tearDown() {
+        context = nil
+        ContextManager.overrideSharedInstance(nil)
+        contextManager = nil
+        super.tearDown()
+    }
+
+    // MARK: - Tests
+
+    func testCanFollowConverationIfReaderTeamTopic() {
```
Comment (emilylaguna) [inline WordPress/WordPressTest/FollowCommentsServiceTests.swift]:
```suggestion
    func testCanFollowConversationIfReaderTeamTopic() {
```

File: WordPress/WordPressTest/FollowCommentsServiceTests.swift [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+        let followCommentsService = FollowCommentsService(post: testPost, remote: remoteMock)!
+
+        // Act
+        let canFollowConversation = followCommentsService.canFollowConversation
+
+        // Asert
+        XCTAssertTrue(canFollowConversation, "Can follow comments on post if the topic is ReaderTeamTopic")
+    }
+
+    func testCanFollowConverationIfWPForTeams() {
```
Comment (emilylaguna) [inline WordPress/WordPressTest/FollowCommentsServiceTests.swift]:
```suggestion
    func testCanFollowConversationIfWPForTeams() {
```

File: WordPress/WordPressTest/FollowCommentsServiceTests.swift [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+        let followCommentsService = FollowCommentsService(post: testPost, remote: remoteMock)!
+
+        // Act
+        let canFollowConversation = followCommentsService.canFollowConversation
+
+        // Asert
+        XCTAssertTrue(canFollowConversation, "Can follow comments on post if blog is marked as a  P2 blog")
+    }
+
+    func testCannotFollowConveration() {
```
Comment (emilylaguna) [inline WordPress/WordPressTest/FollowCommentsServiceTests.swift]:
```suggestion
    func testCannotFollowConversation() {
```

File: WordPress/Classes/ViewRelated/Reader/WPStyleGuide+Reader.swift [hunk 7]
```diff
@@ -281,6 +281,46 @@ extension WPStyleGuide {
...
+        button.setImage(followingIcon.imageWithTintColor(highlightedColor), for: .highlighted)
+        button.imageEdgeInsets = FollowConversationButton.Style.imageEdgeInsets
+        button.contentEdgeInsets = FollowConversationButton.Style.contentEdgeInsets
+
+        // Strings
+        let normalText = FollowConversationButton.Text.followConversationStringForDisplay
+        let selectedText = FollowConversationButton.Text.followingConversationStringForDisplay
+
+        button.setTitle(normalText, for: .normal)
+        button.setTitle(selectedText, for: .selected)
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Reader/WPStyleGuide+Reader.swift]: I know this was done in the `applyReaderFollowButtonStyle` declaration above, but I don't think we should include strings in the style guide. IMO the style guide is for colors, images, and fonts. 

Thoughts?

File: WordPress/WordPressTest/FollowCommentsServiceTests.swift [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+        let remoteMock = ReaderPostServiceRemoteMock()
+        seedBlog(isWPForTeams: false)
+        let testTopic = seedReaderTeamTopic()
+        let testPost = seedReaderPostForTopic(testTopic)
+        let followCommentsService = FollowCommentsService(post: testPost, remote: remoteMock)!
+
+        // Act
+        let canFollowConversation = followCommentsService.canFollowConversation
+
+        // Asert
```
Comment (emilylaguna) [inline WordPress/WordPressTest/FollowCommentsServiceTests.swift]:
```suggestion
        // Assert
```

**Review by emilylaguna — CHANGES_REQUESTED** [review]
Overall this looks good and works functionality wise 👍  

I tested on:

iPhone 11 Pro - 13.3
iPhone 5s - 11.4

I left some comments below, mostly some typos that I found. 

Let me know your thoughts?

File: WordPress/Classes/ViewRelated/Reader/WPStyleGuide+Reader.swift (hunk 7 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Reader/WPStyleGuide+Reader.swift]: Done in dc76de80eb6ddd04bfff19fc360baed7c26641a8

**Review by emilylaguna — APPROVED** [review]
🚢 !

**Pull request merged**